src/cpu/aarch64/jit_uni_reorder.cpp

```diff
@@ -170,7 +170,8 @@ struct jit_uni_reorder_kernel_f32_t : public kernel_t, public jit_generator {
                 || (p.itype == bf16 && p.otype == f32 && mayiuse_bf16()
                         && p.beta == 0.f);
 
-        bool f16_ok = (p.itype != f16 && p.otype != f16)
+        bool is_f16 = (p.itype == f16 || p.otype == f16);
+        bool f16_ok = (p.itype == f32 && p.otype == f16 && p.beta == 0.f)
                 || (p.itype == f16 && p.otype == f32 && p.beta == 0.f);
```

Comment:
(nit) would it make sense to align logic for bf16 and f16 for readability?
- bf16_ok includes the `IMPLICATION(is_bf16, ...)` logic through `|| (p.itype != bf16 && p.otype != bf16)`
-  but f16_ok does not include it and handles it separately with is_f16.

Reply:
@mgouicem This is a good point. I am going to merge this change as-is for now but we have more patches for this file on the way and the logic will be cleaned up there. Thanks.